examples/with-cookie-auth/server.js

```diff
@@ -0,0 +1,78 @@
+const axios = require('axios')
+const express = require('express')
+const cookieParser = require('cookie-parser')
+const next = require('next')
+
+const dev = process.env.NODE_ENV !== 'production'
+
+const AUTHENTICATED_TYPE = 'authenticated'
+const COOKIE_SECRET = 'some deep dark secret'
+const COOKIE_OPTIONS = {
+  // domain: 'YOU_DOMAIN',
+  path: '/',
+  secure: !dev,
+  httpOnly: true,
+  signed: true
+}
+
+// fake serverless authentication
+const authenticate = async (email, password) => {
+  const users = await axios.get('https://jsonplaceholder.typicode.com/users').then(response => response.data)
+  const user = users.find(u => u.email.toLowerCase() === email.toLowerCase())
+  if (user && user.website === password) {
+    return user
+  }
+}
+
+const port = parseInt(process.env.PORT, 10) || 3000
+const app = next({ dev })
+const handle = app.getRequestHandler()
+
+app.prepare()
+  .then(() => {
+    const server = express()
+
+    server.use(express.json())
+
+    server.use(cookieParser(COOKIE_SECRET))
+
+    server.post('/api/login', async (req, res) => {
+      const { email, password } = req.body || {}
+      const user = await authenticate(email, password)
+      if (!user) {
+        return res.status(403).send('Invalid email or password')
+      }
+      const info = {
+        email: user.email,
+        name: user.name,
+        type: AUTHENTICATED_TYPE
+      }
+      res.cookie('token', info, COOKIE_OPTIONS)
+      return res.status(200).json(info)
+    })
+
+    server.post('/api/logout', (req, res) => {
```

Comment:
The API should be separate from the Next.js server

Reply:
It could be. On sites of a certain size, it probably should be.
This is an example. This way, there is no need to fire another server and no CORS problems...